tests/test_utils/test_validators.py

```diff
@@ -267,3 +269,44 @@ class CollectionSchema(BaseSchema):
     assert collection.catalog_id == "byoc-test"
     assert collection.bands == (Band("Band1", (Unit.DN,), (np.float32,)), Band("Band2", (Unit.REFLECTANCE,), (bool,)))
     assert collection.is_byoc is True
+
+
+class DummyFeatureSchema(BaseSchema):
+    mask_like_feature: Optional[Feature]
+    _check_mask_feature = field_validator(
+        "mask_like_feature", restrict_types([FeatureType.MASK, FeatureType.MASK_TIMELESS])
+    )
+
+    temporal_feature: Optional[Feature]
+    _check_temporal_feature = field_validator(
+        "temporal_feature", restrict_types([ftype for ftype in FeatureType if ftype.is_temporal()])
+    )
+
+    feature_3d: Optional[Feature]
+    _check_3d_feature = field_validator(
+        "feature_3d", restrict_types([ftype for ftype in FeatureType if ftype.ndim() == 3])
+    )
+
+
+@pytest.mark.parametrize(
+    "valid_config, invalid_config",
```

Comment:
i would prefer if you split it into two tests, one for valid configurations and one for invalid ones

Reply:
test split. added also a mixed usage of optional and required inputs, so that the restrictor is used in both types of field_validators.